aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -215,6 +215,12 @@
   public static final String S3_CHECKSUM_ENABLED = "s3.checksum-enabled";
   public static final boolean CLIENT_ENABLE_ETAG_CHECK_DEFAULT = false;
 
+  /**
+   * FileIOMetrics implementation class used to report metrics
+   */
+  public static final String S3_FILEIO_METRICS_IMPL = "s3.fileio.metrics-impl";
```

Comment:
What's the benefit of making this dynamic? I thought this was just an adapter that can be used to record metrics in `FileIO`. Shouldn't that be the same and we just customize what accepts the metrics later?

Reply:
I've removed the override properties for now.  We can always add them back when there's a use-case for overriding the default hadoop implementation.